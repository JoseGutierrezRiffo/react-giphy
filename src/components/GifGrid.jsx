import React, { Fragment } from "react";
import GifGridItem from "./GifGridItem";
import { useFetchGifs } from "../hooks/useFetchGifs";

const GifGrid = ({ category }) => {
  const { data:images, loading } = useFetchGifs(category);

  return (
    <Fragment>
      <h3>{category}</h3>
      {loading && <p>Cargando...</p>}
      <div className="card-grid">
        {images.map(image => (
          <GifGridItem key={image.id} image={image} />
        ))}
      </div>
    </Fragment>
  );
};

export default GifGrid;
