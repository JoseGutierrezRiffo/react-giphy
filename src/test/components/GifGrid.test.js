import React from "react";
import { shallow } from "enzyme";
import GifGrid from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

const categoria = "Morty";

describe("prueba con <GifGrid />", () => {
  test("debe mostrar el componente", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    });
    const wrapper = shallow(<GifGrid category={categoria} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("debe mostrar items cuando se cargan ls imagenes useFetchGifs", () => {
    const gifs = [
      {
        id: "JqA8xufDs9ojSo9Cfjy7Zn",
        title: "titulo de la imagen 1",
        urlImg: "http://localhost/imagen1.png"
      },
      {
        id: "KqA8xufDs9ojSo9Cfjy7Zn",
        title: "titulo de la imagen 2",
        urlImg: "http://localhost/imagen2.png"
      },
      {
        id: "LqA8xufDs9ojSo9Cfjy7Zn",
        title: "titulo de la imagen 3",
        urlImg: "http://localhost/imagen3.png"
      }
    ];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false
    });
    const wrapper = shallow(<GifGrid category={categoria} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("p").exists()).toBe(false);
    expect(wrapper.find("GifGridItem").length).toBe(gifs.length);
  });
});
