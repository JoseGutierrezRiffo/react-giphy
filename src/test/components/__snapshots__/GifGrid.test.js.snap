// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prueba con <GifGrid /> debe mostrar el componente 1`] = `
<Fragment>
  <h3>
    Morty
  </h3>
  <p>
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`prueba con <GifGrid /> debe mostrar items cuando se cargan ls imagenes useFetchGifs 1`] = `
<Fragment>
  <h3>
    Morty
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="JqA8xufDs9ojSo9Cfjy7Zn"
      key="JqA8xufDs9ojSo9Cfjy7Zn"
      title="titulo de la imagen 1"
      urlImg="http://localhost/imagen1.png"
    />
    <GifGridItem
      id="KqA8xufDs9ojSo9Cfjy7Zn"
      key="KqA8xufDs9ojSo9Cfjy7Zn"
      title="titulo de la imagen 2"
      urlImg="http://localhost/imagen2.png"
    />
    <GifGridItem
      id="LqA8xufDs9ojSo9Cfjy7Zn"
      key="LqA8xufDs9ojSo9Cfjy7Zn"
      title="titulo de la imagen 3"
      urlImg="http://localhost/imagen3.png"
    />
  </div>
</Fragment>
`;
